src/box/sql/build.c

```diff
@@ -3344,24 +3344,20 @@ int
 sql_fieldno_by_name(struct Parse *parse_context, struct Expr *field_name,
 		    uint32_t *fieldno)
 {
-	struct space_def *def = parse_context->create_table_def.new_space->def;
+	const struct space *space = parse_context->create_table_def.new_space;
 	struct Expr *name = sqlExprSkipCollate(field_name);
 	if (name->op != TK_ID) {
 		diag_set(ClientError, ER_INDEX_DEF_UNSUPPORTED, "Expressions");
 		parse_context->is_aborted = true;
 		return -1;
 	}
-	uint32_t i;
-	for (i = 0; i < def->field_count; ++i) {
-		if (strcmp(def->fields[i].name, name->u.zToken) == 0)
-			break;
-	}
-	if (i == def->field_count) {
+	uint32_t id = sql_fieldno_by_expr(space, name);
```

Comment:
We can use space_def here instead of space, but I used space because of space always available where these functions are called.

In general, I once tried to replace "ephemeral" spaces used during CREATE TABLE/ADD COLUMN by space_def, but it is quite troublesome since some parts of the spaces also used in these statements, for example indexes.